src/keyboard-modality.js

```diff
@@ -9,11 +10,40 @@ document.addEventListener("DOMContentLoaded", function() {
                                       "textarea",
                                       "[role=textbox]",
                                       "[supports-modality=keyboard]"].join(","),
-        matcher = getMatcher();
+        matcher = (function () {
+			var el = document.body;
+			if (el.matchesSelector)
+				return el.matchesSelector;
+			if (el.webkitMatchesSelector)
+				return el.webkitMatchesSelector;
+			if (el.mozMatchesSelector)
+				return el.mozMatchesSelector;
+			if (el.msMatchesSelector)
+				return el.msMatchesSelector;
+			console.error("Couldn't find any matchesSelector method on document.body.");
+		}()),
+        disableFocusRingByDefault = function () {
+			var css = "body:not([modality=keyboard]) :focus { outline: none; }",
+				head = document.head || document.getElementsByTagName("head")[0],
+				style = document.createElement("style");
+
+			style.type = "text/css";
+			style.id = "disable-focus-ring";
+			if (style.styleSheet) {
+				style.styleSheet.cssText = css;
+			} else {
+				style.appendChild(document.createTextNode(css));
+			}
+
+			head.insertBefore(style, head.firstChild);
+		},
+		focusTriggersKeyboardModality = function (el) {
+			return matcher.call(el, keyboardModalityWhitelist);
```

Comment:
good idea, I will except it does seem different as that is async on capture and doesn't cancel or anything -- will it actually prevent something else from running?  I don't think so... I can add check, it's the right thing to do either way :)